pacman/model/partitioner_splitters/splitter_external_device.py

```diff
@@ -48,23 +48,22 @@ def set_governed_app_vertex(self, app_vertex):
 
         if isinstance(app_vertex, ApplicationFPGAVertex):
             # This can have multiple FPGA connections per board
-            if app_vertex.incoming_fpga_connections:
-                for fpga in app_vertex.incoming_fpga_connections:
-                    for i in range(app_vertex.n_machine_vertices_per_link):
-                        label = (
-                            f"Incoming Machine vertex {i} for"
-                            f" {app_vertex.label}"
-                            f":{fpga.fpga_id}:{fpga.fpga_link_id}"
-                            f":{fpga.board_address}:{fpga.chip_coords}")
-                        vertex_slice = app_vertex.get_incoming_slice_for_link(
-                            fpga, i)
-                        vertex = MachineFPGAVertex(
-                            fpga.fpga_id, fpga.fpga_link_id,
-                            fpga.board_address, fpga.chip_coords, label=label,
-                            app_vertex=app_vertex, vertex_slice=vertex_slice,
-                            incoming=True, outgoing=False)
-                        self.__incoming_vertices.append(vertex)
-                        self.__incoming_slices.append(vertex_slice)
+            for fpga in app_vertex.incoming_fpga_connections:
```

Comment:
Is `app_vertex.incoming_fpga_connections` guaranteed to be iterable at this point? (If it is, this is a good change as it reduces the cyclomatic complexity.)

Reply:
That is how it is described and also how it is implemented.

See: https://github.com/SpiNNakerManchester/PACMAN/blob/664f336254b1973301aad2f93db1f7a118269417/pacman/model/graphs/application/application_fpga_vertex.py#L103